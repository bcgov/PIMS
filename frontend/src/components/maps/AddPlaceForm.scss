.modal-body {
    display: flex;
  p {
      widows: 324px;
      height: 37px;
  }
  img {
      margin-right: 21px;
  }
}
.add-place {
  width: 212px;
  height: 180px;
  margin: 25px;

  form {
    font-family: ‘Noto Sans’, Verdana, Arial, sans-serif;
    font-size: 18px;
  }

  h1 {
    text-align: left;
    height: 19px;
    margin: 0px;
    font-family: Arial;
    font-size: 16px;
    color: #606060;
    text-decoration: none solid rgb(96, 96, 96);
  }

  .button {
    display: block;
    width: 82px;
    box-sizing: border-box;
    margin: 0;
    padding: 10px 16px;
    width: 100%;
    outline: 0;
    border: none;
    border-color: none;
    border-radius: 0;
    color: #fff;
    font-family: Arial;
    text-align: center;
    text-decoration: none;
    font-size: 16px;
    line-height: 1.3333333;
    cursor: pointer;
    border-radius: 50% !important;
  }

  .BC-Gov-PrimaryButton {
    background-color: #38598a;
    border-radius: 4px;
    color: white;
    display: inline-block;
    margin-left: 48px;
  }

  .button:hover {
    background: rgba(54, 50, 119, 0.8);
  }

  .button:focus {
    border-color: #0074c2;
    box-shadow: 0 0 8px rgba(0, 116, 194, 0.6);
  }
}
