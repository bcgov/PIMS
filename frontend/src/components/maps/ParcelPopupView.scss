@import "../../fonts.scss";
.parcelPopup {
  min-width: 211px;
  max-width: 250px;
  box-shadow: 1px 2px 2px;
  text-align: left;
  padding: 12px;
  .row {
    flex-direction: column;
  }
  .row .col :first-child {
    margin-top: 0px;
  }

  .label {
    font-weight: 700;
  }

  .list-group {
    margin-top: 6px;
  }
  .list-group-item {
    padding: 0;
    border: none;
    font-family: "BCSans" Fallback, sans-serif;
    font-size: 12px;
  }
  p {
      margin: 0;
  }
  .alert {
    margin: 0;
  }
}
