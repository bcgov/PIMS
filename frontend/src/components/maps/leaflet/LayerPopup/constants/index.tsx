import * as React from 'react';

export const MUNICIPALITY_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP';
export const PARCELS_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW/wfs?service=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW';

export const parcelLayerPopupConfig = {
  PARCEL_NAME: { label: 'Parcel Name:', display: (data: any) => data.PARCEL_NAME },
  PIN: { label: 'Parcel PIN:', display: (data: any) => data.PIN },
  PID: { label: 'Parcel PID:', display: (data: any) => data.PID },
  PLAN_NUMBER: { label: 'Plan Number:', display: (data: any) => data.PLAN_NUMBER },
  PARCEL_CLASS: { label: 'Parcel Class:', display: (data: any) => data.PARCEL_CLASS },
  MUNICIPALITY: { label: 'Municipality:', display: (data: any) => data.MUNICIPALITY },
  REGIONAL_DISTRICT: {
    label: 'Regional District:',
    display: (data: any) => data.REGIONAL_DISTRICT,
  },
  FEATURE_AREA_SQM: {
    label: 'Area:',
    display: (data: any) => (
      <>
        {data.FEATURE_AREA_SQM} m<sup>2</sup>
      </>
    ),
  },
};

export const municipalityLayerPopupConfig = {
  ADMIN_AREA_GROUP_NAME: {
    label: 'Administration Area:',
    display: (data: any) => `${data.ADMIN_AREA_GROUP_NAME} (${data.CHANGE_REQUESTED_ORG})`,
  },
};
