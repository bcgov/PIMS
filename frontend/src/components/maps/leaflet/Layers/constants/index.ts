export const MUNICIPALITY_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP';
export const PARCELS_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW/wfs?service=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW';

export const parcelLayerPopupConfig = {
  PIN: { label: 'Parcel PIN:', display: (data: any) => data.PIN },
  PID: { label: 'Parcel PID:', display: (data: any) => data.PID },
  REGIONAL_DISTRICT: {
    label: 'Regional District:',
    display: (data: any) => data.REGIONAL_DISTRICT,
  },
};

export const municipalityLayerPopupConfig = {
  ADMIN_AREA_NAME: { label: 'Area Name:', display: (data: any) => data.ADMIN_AREA_NAME },
  ADMIN_AREA_ABBREVIATION: {
    label: 'Area Abbrevation:',
    display: (data: any) => data.ADMIN_AREA_ABBREVIATION,
  },
  ADMIN_AREA_GROUP_NAME: {
    label: 'Area Group Name:',
    display: (data: any) => `${data.ADMIN_AREA_GROUP_NAME} (${data.CHANGE_REQUESTED_ORG})`,
  },
};
