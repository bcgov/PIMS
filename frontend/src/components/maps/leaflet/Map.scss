@import '../../../variables';
.leaflet-container {
  /* Subtract the header, nav-bar and footer sizes */
  height: calc(
    100vh - #{$header-height} - #{$navbar-height} - #{$mapfilter-height} - #{$footer-height}
  );
  // margin-top: 10px;
  // padding-top: 20px;
  width: 100%;
  margin: auto;

  .leaflet-popup-content-wrapper {
    border-radius: 0;
  }
  .leaflet-popup-content {
    margin: 0px;
    .list-group {
      .list-group-item {
        border: none;
        padding: 0.5rem 1.25rem;
      }
    }
  }
}

.leaflet-control-layers {
  &.leaflet-control-layers-expanded {
    border-radius: 0px;
  }
  .leaflet-control-layers-overlays {
    display: flex;
    flex-direction: column;
    padding: 10px;
    width: 150px;
    label {
      display: flex;
    }
  }
}
