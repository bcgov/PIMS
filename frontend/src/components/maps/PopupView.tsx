import * as React from 'react';
import { IBuilding, IParcel } from 'actions/parcelsActions';
import { ParcelPopupView } from './ParcelPopupView';
import { BuildingPopupView } from './BuildingPopupView';

export type IPopupViewProps = {
  propertyTypeId: 0 | 1; // 0 = Parcel, 1 = Building
  propertyDetail: IParcel | IBuilding | null;
  disabled?: boolean;
};

export const PopupView: React.FC<IPopupViewProps> = ({
  propertyTypeId,
  propertyDetail,
  disabled,
}) => {
  if (propertyTypeId === 0) {
    return <ParcelPopupView disabled={disabled} parcel={propertyDetail as IParcel} />;
  }
  if (propertyTypeId === 1) {
    return <BuildingPopupView building={propertyDetail as IBuilding} />;
  }
  return null;
};
