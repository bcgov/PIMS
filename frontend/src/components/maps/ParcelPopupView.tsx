import * as React from 'react';
import { ListGroup, Container } from 'react-bootstrap';
import { IParcel } from 'actions/parcelsActions';
import { Alert, Row, Col } from 'react-bootstrap';
import { Label } from 'components/common/Label';
import './ParcelPopupView.scss';
import { Link } from 'react-router-dom';
import { EvaluationKeys } from '../../constants/evaluationKeys';
import useKeycloakWrapper from 'hooks/useKeycloakWrapper';

export interface IParcelDetailProps {
  parcel: IParcel | null;
  disabled?: boolean;
}

export const ParcelPopupView = (props: IParcelDetailProps | null) => {
  const parcelDetail: IParcel | null | undefined = props?.parcel;
  const keycloak = useKeycloakWrapper();

  return (
    <Container className="parcelPopup" fluid={true}>
      {!parcelDetail ? (
        <Alert variant="danger">Failed to load parcel details.</Alert>
      ) : (
        <Row>
          <Col>
            <ListGroup>
              <ListGroup.Item>
                <Label>Assessed Value: </Label>$
                {parcelDetail?.evaluations
                  ?.find(e => e.key === EvaluationKeys.Assessed)
                  ?.value?.toLocaleString('en-US', { minimumFractionDigits: 2 })}
              </ListGroup.Item>
            </ListGroup>
            <ListGroup>
              <ListGroup.Item>
                <div>
                  <Label>Address: </Label>
                  {parcelDetail?.address?.line1}
                </div>
                <div>
                  {parcelDetail?.address?.city}, {parcelDetail?.address?.province}{' '}
                  {parcelDetail?.address?.postal}
                </div>
              </ListGroup.Item>
              <ListGroup.Item>
                <Label>PID: </Label> {parcelDetail?.pid}
              </ListGroup.Item>
              {parcelDetail?.projectNumber && (
                <ListGroup.Item>
                  <Label>RAEG or SPP:</Label> {parcelDetail?.projectNumber}
                </ListGroup.Item>
              )}
            </ListGroup>
            <ListGroup>
              <ListGroup.Item>
                <Label>Agency: </Label>
                {parcelDetail?.agency}
              </ListGroup.Item>
              <ListGroup.Item>
                <Label>Classification: </Label>
                {parcelDetail?.classification}
              </ListGroup.Item>
            </ListGroup>
          </Col>
          <Col>
            {parcelDetail?.id &&
              !props?.disabled &&
              (!keycloak.hasAgency(parcelDetail?.agencyId) ? (
                <Link to={`/submitProperty/${parcelDetail?.id}?disabled=true`}>View</Link>
              ) : (
                <Link to={`/submitProperty/${parcelDetail?.id}`}>Update</Link>
              ))}
          </Col>
        </Row>
      )}
    </Container>
  );
};
