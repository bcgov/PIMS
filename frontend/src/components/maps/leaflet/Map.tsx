import './Map.scss';

import React, { useRef, useState, useEffect } from 'react';
import axios from 'axios';
import { LatLngBounds, LeafletMouseEvent } from 'leaflet';
import { Map as LeafletMap, TileLayer, Marker, Popup } from 'react-leaflet';
import { IParcel, IParcelDetail, IProperty, IPropertyDetail } from 'actions/parcelsActions';
import { ParcelPopupView } from '../ParcelPopupView';
import { Container, Row } from 'react-bootstrap';
import MapFilterBar, { MapFilterChangeEvent } from '../MapFilterBar';
import { ILookupCode } from 'actions/lookupActions';
import BasemapToggle, { BasemapToggleEvent, BaseLayer } from '../BasemapToggle';
import { decimalOrNull, floatOrNull } from 'utils';
import { PopupView } from '../PopupView';

export type MapViewportChangeEvent = {
  bounds: LatLngBounds | null;
  filter?: {
    address: string;
    /** comma-separated list of agencies to filter by */
    agencies: string | null;
    classificationId: number | null;
    minLotSize: number | null;
    maxLotSize: number | null;
  };
};

type MapProps = {
  lat: number;
  lng: number;
  zoom: number;
  properties: IProperty[];
  agencies: ILookupCode[];
  propertyClassifications: ILookupCode[];
  lotSizes: number[];
  selectedProperty?: IPropertyDetail | null;
  onMarkerClick?: (obj: IProperty) => void;
  onMarkerPopupClose?: (obj: IPropertyDetail) => void;
  onViewportChanged?: (e: MapViewportChangeEvent) => void;
  onMapClick?: (e: LeafletMouseEvent) => void;
  disableMapFilterBar?: boolean;
  disabled?: boolean;
};

const Map: React.FC<MapProps> = props => {
  // props
  const {
    properties,
    selectedProperty,
    onMarkerClick,
    onMarkerPopupClose,
    onViewportChanged,
  } = props;
  const mapRef = useRef<LeafletMap>(null);
  const [mapFilter, setMapFilter] = useState<MapFilterChangeEvent>({
    address: '',
    agencies: '',
    classificationId: '',
    minLotSize: '',
    maxLotSize: '',
  });
  const [baseLayers, setBaseLayers] = useState<BaseLayer[]>([]);
  const [activeBasemap, setActiveBasemap] = useState<BaseLayer | null>(null);

  if (props.disabled) {
    const map = mapRef.current?.leafletElement;
    if (map) {
      map.dragging.disable();
      map.touchZoom.disable();
      map.doubleClickZoom.disable();
      map.scrollWheelZoom.disable();
      map.boxZoom.disable();
      map.keyboard.disable();
      if (map.tap) map.tap.disable();
    }
  }
  // --- Internal functions and event handlers
  const getBounds = () => {
    if (!mapRef.current) {
      return null;
    }
    return mapRef.current?.leafletElement.getBounds();
  };

  const handleViewportChange = () => {
    const bounds = getBounds();
    const { address, agencies, classificationId, minLotSize, maxLotSize } = mapFilter;
    const e: MapViewportChangeEvent = {
      bounds,
      filter: {
        address,
        agencies: agencies,
        classificationId: decimalOrNull(classificationId),
        minLotSize: floatOrNull(minLotSize),
        maxLotSize: floatOrNull(maxLotSize),
      },
    };
    onViewportChanged?.(e);
  };

  const handleMapFilterChange = (e: MapFilterChangeEvent) => {
    setMapFilter(e);
  };

  const handleBasemapToggle = (e: BasemapToggleEvent) => {
    const { previous, current } = e;
    setBaseLayers([current, previous]);
    setActiveBasemap(current);
  };

  // --- Effects AKA network requests
  useEffect(() => {
    handleViewportChange();
  }, [mapFilter]);

  useEffect(() => {
    // fetch GIS base layers configuration from /public folder
    axios.get('/basemaps.json').then(result => {
      setBaseLayers(result.data?.basemaps);
      setActiveBasemap(result.data?.basemaps?.[0]);
    });
  }, []);

  // we need to namespace the keys as IDs are not enough here.
  // the same ID could be found on both the parcel collection and building collection
  const generateKey = (p: IProperty) => `${p.propertyTypeId === 0 ? 'parcel' : 'building'}-${p.id}`;

  const renderMarker = (p: IProperty) => (
    <Marker
      key={generateKey(p)}
      position={[p.latitude, p.longitude]}
      onClick={(e: any) => onMarkerClick?.(p)}
    />
  );

  const renderPopup = (item: IPropertyDetail) => {
    const { propertyTypeId, parcelDetail } = item;
    if (!parcelDetail) {
      return null;
    }
    return (
      <Popup
        position={[parcelDetail.latitude, parcelDetail.longitude]}
        offset={[0, -25]}
        onClose={() => !props?.disabled && onMarkerPopupClose?.(item)}
        closeButton={!props.disabled}
      >
        <PopupView
          propertyTypeId={propertyTypeId}
          propertyDetail={parcelDetail}
          disabled={props.disabled}
        />
      </Popup>
    );
  };

  return (
    <Container fluid={true}>
      <Row>
        {!props.disableMapFilterBar ? (
          <MapFilterBar
            agencyLookupCodes={props.agencies}
            propertyClassifications={props.propertyClassifications}
            lotSizes={props.lotSizes}
            onFilterChange={handleMapFilterChange}
          />
        ) : null}
        {baseLayers?.length > 0 && (
          <BasemapToggle baseLayers={baseLayers} onToggle={handleBasemapToggle} />
        )}
        <LeafletMap
          ref={mapRef}
          center={[props.lat, props.lng]}
          zoom={props.zoom}
          whenReady={handleViewportChange}
          onViewportChanged={handleViewportChange}
          onpreclick={props.onMapClick}
          closePopupOnClick={!props.disabled}
        >
          {activeBasemap && (
            <TileLayer attribution={activeBasemap.attribution} url={activeBasemap.url} />
          )}
          {properties && properties.map(renderMarker)}
          {selectedProperty && renderPopup(selectedProperty)}
        </LeafletMap>
      </Row>
    </Container>
  );
};

export default Map;
