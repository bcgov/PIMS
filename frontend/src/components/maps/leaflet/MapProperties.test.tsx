import React from 'react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ParcelPopupView } from 'components/maps/ParcelPopupView';
import { IParcel, IProperty, IParcelDetail } from 'actions/parcelsActions';
import Map from './Map';
import { Marker } from 'react-leaflet';
import { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import * as reducerTypes from 'constants/reducerTypes';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';
import { PopupView } from '../PopupView';
import { Provider } from 'react-redux';

Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureMockStore([thunk]);

const store = mockStore({
  [reducerTypes.LOOKUP_CODE]: { lookupCodes: [] },
});

// This will spoof the active parcel (the one that will populate the popup details)
const mockDetails: IParcelDetail = {
  propertyTypeId: 0,
  parcelDetail: {
    id: 1,
    pid: '000-000-000',
    pin: '',
    statusId: 0,
    classificationId: 0,
    zoning: '',
    zoningPotential: '',
    agencyId: 0,
    latitude: 48,
    longitude: 123,
    propertyStatus: 'active',
    classification: 'Core Operational',
    description: 'test',
    isSensitive: false,
    evaluations: [
      {
        assessedValue: 1000000,
        estimatedValue: 0,
        fiscalYear: 2019,
        netBookValue: 0,
        appraisedValue: 0,
      },
    ],
    address: {
      line1: '1234 mock Street',
      line2: 'N/A',
      city: 'Victoria',
      province: 'BC',
      postal: 'V1V1V1',
    },
    landArea: 'unknown',
    landLegalDescription: 'test',
    buildings: [],
    agency: 'FIN',
  },
};

// To check for alert message
const emptyDetails = null;

// This mocks the parcels of land a user can see - should be able to see 2 markers
const mockParcels = [
  { id: 1, latitude: 48, longitude: 123 },
  { id: 2, latitude: 50, longitude: 133 },
] as IProperty[];
const noParcels = [] as IProperty[];

const history = createMemoryHistory();

// Check that the markers have correct position given the mock parcel 1 above
it('Renders the marker in correct position', () => {
  const component = mount(
    <Provider store={store}>
      <Router history={history}>
        <Map
          lat={48.43}
          lng={-123.37}
          zoom={14}
          properties={mockParcels}
          selectedProperty={mockDetails}
          agencies={[]}
          propertyClassifications={[]}
          lotSizes={[]}
          onMarkerClick={jest.fn()}
        />
      </Router>
    </Provider>,
  );
  const marker = component.find(Marker).first();
  expect(marker.prop('position')).toStrictEqual([48, 123]);
});

// Ensure no markers are rendered when there are no parcels
it('Should render 0 markers when there are no parcels', () => {
  const component = mount(
    <Router history={history}>
      <Map
        lat={48.43}
        lng={-123.37}
        zoom={14}
        properties={noParcels}
        selectedProperty={emptyDetails}
        agencies={[]}
        propertyClassifications={[]}
        lotSizes={[]}
        onMarkerClick={jest.fn()}
      />
    </Router>,
  );
  const marker = component.find(Marker);
  expect(marker.length).toBe(0);
});

// 2 parcels in mock data, check to see 2 markers are created
it('Marker for each parcel is created', () => {
  const component = mount(
    <Provider store={store}>
      <Router history={history}>
        <Map
          lat={48.43}
          lng={-123.37}
          zoom={14}
          properties={mockParcels}
          selectedProperty={mockDetails}
          agencies={[]}
          propertyClassifications={[]}
          lotSizes={[]}
          onMarkerClick={jest.fn()}
        />
      </Router>
    </Provider>,
  );
  const marker = component.find(Marker);
  expect(marker.length).toBe(2);
});

// When marker is clicked function to load the details should be called
it('Loads parcel details on click', () => {
  const onParcelClick = jest.fn();
  const component = shallow(
    <Map
      lat={48.43}
      lng={-123.37}
      zoom={14}
      properties={mockParcels}
      selectedProperty={mockDetails}
      agencies={[]}
      propertyClassifications={[]}
      lotSizes={[]}
      onMarkerClick={onParcelClick}
    />,
  );
  const marker = component.find(Marker).first();
  marker.simulate('click');
  expect(onParcelClick).toBeCalledTimes(1);
});

// Check that error message is displayed on null details
it('Displays proper message when no details loaded', () => {
  const { getByText } = render(
    <Provider store={store}>
      <Router history={history}>
        <ParcelPopupView parcel={emptyDetails} />
      </Router>
    </Provider>,
  );
  const alert = getByText('Failed to load parcel details.');
  expect(alert).toBeTruthy();
});

it('ParcelPopupView renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router history={history}>
          <PopupView
            propertyTypeId={mockDetails.propertyTypeId}
            propertyDetail={mockDetails.parcelDetail}
          />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
