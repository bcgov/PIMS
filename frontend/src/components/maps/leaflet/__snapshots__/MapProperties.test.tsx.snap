// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapProperties View ParcelPopupView renders correctly 1`] = `
<div
  className="parcelPopup container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <div
          className="pid list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          000-000-000
        </div>
        <div
          className="name list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          test name
        </div>
        <div
          className="description list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          test
        </div>
      </div>
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <div
          className="list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <p
            className="label"
          >
            Assessed Value:
          </p>
          $
        </div>
      </div>
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <div
          className="list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <div>
            1234 mock Street
          </div>
          <div>
            
            , 
             
            V1V1V1
          </div>
        </div>
        
      </div>
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <div
          className="list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <p
            className="label"
          >
            Agency:
          </p>
          FIN
        </div>
        <div
          className="list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <p
            className="label"
          >
            Classification: 
          </p>
          Core Operational
        </div>
      </div>
    </div>
  </div>
  <div
    className="menu row"
  >
    <div
      className="col"
    >
      <a
        href="/mapview/1?disabled=true&loadDraft=false&sidebar=true"
        onClick={[Function]}
      >
        View
      </a>
      <a
        href="/mapview/1?disabled=false&loadDraft=false&sidebar=true"
        onClick={[Function]}
      >
        Update
      </a>
    </div>
  </div>
</div>
`;
